app/presenters/brick_presenter.rb

```diff
@@ -3,7 +3,7 @@ class << self
     def list(bricks)
       bricks.map do |attributes|
         %w[devices partitions pv].each do |key|
-          attributes[key] = JSON.parse(attributes[key] || '[]')
+          attributes[key] = JSON.parse(attributes[key] || '[]') rescue {}
```

Comment:
So JSON.parse call has `|| []` already. Doesnt that suffice to make sure if the key missing set the value as `[]`?

Reply:
attributes[key] do not return any boolean value so using '||' for this won't solve any issue. Ideally, the line should be like: 
attributes[key] = JSON.parse(attributes[key]) rescue  []

we don't need '||' here, because if attributes[key] is not '{}' or '[]' it will raise exception, or condition ("||") won't handle anything.

@shtripat comment  please

Reply:
Looks fine to me. @shirshendu ^^^

Reply:
@GowthamShanmugam This is not how `||` works in ruby. Not just boolean, also when `attributes[key]` is `nil`, the parsed string will be `[]`. This is the case because in all three of  `attributes['devices']` or `attributes['partitions']` or  `attributes['pv']`, the default string being received from the etcd store was a JSON string.
I don't see the need for a `rescue` here.
Can you give a sample value of attributes[key] which raises an exception?

Reply:
>the exact semantics of || are:
>- if first expression is not nil or false, return it
>- if first expression is nil or false, return the second expression